alpha/lib/model/DeliveryProfileLive.php

```diff
@@ -397,6 +401,46 @@ protected function getRenderer($flavors)
 		return $renderer;
 	}
 	
+	protected function retrieveStreamIdsByBitrate ($streams)
+	{
+		$streamIds = array();
+		if (!$streams || !count($streams))
+		{
+			return;
+		}
+		elseif (!$this->getDynamicAttributes()->getMaxBitrate() && !$this->getDynamicAttributes()->getMinBitrate())
+		{
+			return;
+		}
+		
+		foreach ($streams as $stream)
+		{
+			/* @var $stream kLiveStreamParams */
+			if ($this->getDynamicAttributes()->getMinBitrate() && $stream->getBitrate()/1024 < $this->getDynamicAttributes()->getMinBitrate())
+			{
+				continue;
+			}
+			if ($this->getDynamicAttributes()->getMaxBitrate() && $stream->getBitrate()/1024 > $this->getDynamicAttributes()->getMaxBitrate())
+			{
+				continue;
+			}
+			
+			$streamIds[] = $stream->getFlavorId();
+			
+			if(count($this->getDynamicAttributes()->getFlavorParamIds()))
+			{
+				$streamIds = array_intersect($streamIds, $this->getDynamicAttributes()->getFlavorParamIds());
+			}
+		}
+		if (!count($streamIds))
+		{
+			// If the stream info is available to us, min and/or max bitrate params were passed, and no streams comply with them - throw error
+			KExternalErrors::dieError(KExternalErrors::FLAVOR_NOT_FOUND, "Entry [". $this->getDynamicAttributes()->getEntryId() ."] has no valid streams");
+		}
+		
+		$this->params->setFlavorParamIds($streamIds);
```

Comment:
All the rest of the places you use $this->getDynamicAttributes() to reach the params, for consistency I suggest to change all the references to be the same.